src/app/modules/schedule/reducer.ts

```diff
@@ -70,6 +70,18 @@ const reducer = createReducer<ScheduleState>({
   },
   SCHEDULE_PUBLISH_VIEW_SHOW_WORKSECTION: (state, showWorkSections) => {
     return { ...state, showWorkSections };
+  },
+  SCHEDULE_SHIFT_SELECT_ALL: (state, shiftIds) => {
+    const { selectedShiftIds = [] } = state;
+
+    let newSelectedShiftIds = selectedShiftIds.slice();
```

Comment:
if we change this to `SET_SELECTED_SHIFTS` then this reducer becomes much simpler.